providers/storage/mongodocstore.js

```diff
@@ -59,19 +59,30 @@ class MongoDocStore {
     });
   }
 
-  listDocuments(pattern) {
-    // TODO implement
-    return Q([]);
+  list(type) {
+    return this.db.collection(type).find({}, { '_metadata': 1 }).toArray().then(docs => {
+      return docs.map(doc => {
+        const metadata = doc._metadata;
+        Object.keys(metadata).forEach(key => {
+          if (key !== key.toLowerCase()) {
+            metadata[key.toLowerCase()] = metadata[key];
+            delete metadata[key];
+          }
+        });
+        metadata.urn = metadata.links.self.href;
+        return metadata;
+      })
+    });
   }
 
   delete(type, url) {
-    // TODO implement
-    return Q(true);
+    return this.db.collection(type).deleteOne({ $or: [{ '_metadata.url': url }, { '_metadata.links.self.href': url }] }).then(result => {
```

Comment:
same comment here about the or and use of href.  if this is prevalent, can we factor out the query object and share it across all the functions?